prog/weaver/main.go

```diff
@@ -427,6 +427,12 @@ func main() {
 	}
 	checkFatal(router.CreateRestartSentinel())
 
+	// Slightly ugly that plugin status is separated from the plugin.Start call
```

Comment:
I'd suggest to slightly refactor the plugin package:

* Introduce the `func NewPlugin(...) *Plugin` function.
* Create the `func (p *Plugin) Start()` method.
* Create the `func (p *Plugin) Status() Status` method.

The change would allow us to pass the `*Plugin` instance instead of  `pluginStatus`.